src/hooks/useDisputes.js

```diff
@@ -61,25 +63,46 @@ export default function useDisputes() {
   }, [disputesPhases, disputes, disputesPhasesKey, error]) // eslint-disable-line react-hooks/exhaustive-deps
 }
 
+/**
+ * Hook for single dispute
+ * @param {String} disputeId Id of the dispute
+ * @returns {Array} First element the dispute, second is the fetching indicator,
+ * third might be the error from the graph or an ipfs error,
+ * third a boolean that indicates if the error is from the graph since we need to handle in a different way
+ * in the dispute detail, the timeline can not be displayed if is a graph error but can if the error is from ipfs
+ */
 export function useDispute(disputeId) {
   const courtConfig = useCourtConfig()
   const now = useNow() // TODO: use court clock
-  const { dispute, fetching } = useSingleDisputeSubscription(disputeId)
-  const disputeProcessed = useProcessedDispute(dispute, fetching)
+  const {
+    dispute,
+    fetching: graphFetching,
+    error: graphError,
+  } = useSingleDisputeSubscription(disputeId)
+
+  const graphErrorMessage = graphError ? graphError.message : ''
+  const disputeProcessed = useProcessedDispute(dispute, graphFetching)
+  const error = graphErrorMessage || disputeProcessed.error
 
   const disputePhase = getPhaseAndTransition(dispute, courtConfig, now)
   const disputePhaseKey = disputePhase
     ? convertToString(Object.values(disputePhase)[0])
     : ''
 
   return useMemo(() => {
-    return {
-      dispute: {
-        ...disputeProcessed,
-        ...disputePhase,
-      },
-    }
-  }, [disputeProcessed, disputePhaseKey]) // eslint-disable-line react-hooks/exhaustive-deps
+    const fetching = graphFetching || disputeProcessed.fetching
+    return [
+      dispute
+        ? {
+            ...disputeProcessed,
+            ...disputePhase,
+          }
+        : null,
+      fetching,
+      error,
+      !!graphErrorMessage,
+    ]
+  }, [disputeProcessed, disputePhaseKey, error, graphFetching]) // eslint-disable-line react-hooks/exhaustive-deps
 }
 
 function useProcessedDispute(dispute, fetching) {
```

Comment:
I'd like to refactor this to how it was implemented before (without setting a default state of `DISPUTE_PROCESSED_DEFAULT` object) because logically if the dispute does not exist it could generate more confusion than the benefits it provides and we could have race conditions between when the dispute is done fetching from the graph and it exists,  and we are still fetching from IPFS.

We could set the default state as `null` and calculate if the dispute is fetching under this condiitons:

- we are still fetching the data from the subgraph. 
- we are done fetching form subgraph
    - if the dispute exists then we are stil fetching if `disputeProcessed != null`
    - if dispute does not exists we just return

Also in this way we separate the fetching logic from this hook.

Reply:
cc @sohkai as he suggested to use the default state

Reply:
The new version looks good! And agreed, having it be `null` makes it more clear it's still being fetched.

Two notes:

1. We forgot to use the `cancel` pattern in `useProcessedDispute()`
2. Are we handling errors returned from `useProcessedDispute()` correctly?

Reply:
> We forgot to use the cancel pattern in useProcessedDispute()

Ah yes good point!

> Are we handling errors returned from useProcessedDispute() correctly?

Yes, the handling is made here https://github.com/aragon/court-dashboard/blob/55271e90738eebf71e192f56c4d95f0cbc7757cf/src/hooks/useDisputes.js#L81

Will do a PR with the cancel pattern.